src/app/core/services/scheduling.service.ts

```diff
@@ -245,9 +247,18 @@ export class SchedulingService {
           repeatQ.unit,
           repeatQ.unitsFromZero[i]
         )
-        const idx = indexOffset + tmpScheduleAll.length
-        const task = this.taskBuilder(idx, assessment, taskDate)
-        if (task.timestamp > today.getTime()) {
+
+        if (taskDate.getTime() > today.getTime()) {
+          const idx = indexOffset + tmpScheduleAll.length
+          const task = this.taskBuilder(
+            idx,
+            assessment,
+            taskDate,
+            this.prevSchedule ? this.prevSchedule[idx].completed : false,
+            this.prevSchedule
+              ? this.prevSchedule[idx].reportedCompletion
+              : false
```

Comment:
Change this condition to be based on comparing todays completed questionnaires (from midnight) on the basis of assessment name and timestamp

Reply:
Made the changes.